src/Test/StrictCheck/Shaped.hs

```diff
@@ -126,7 +126,7 @@ class Typeable a => Shaped (a :: *) where
   -- | The @Shape@ of an @a@ is a type isomorphic to the outermost level of
   -- structure in an @a@, parameterized by the functor @f@, which is wrapped
   -- around any fields (of any type) in the original @a@.
-  type Shape a :: (* -> *) -> *
+  type Shape a = (result :: (* -> *) -> *) | result -> a
```

Comment:
I like this, especially the way it aids type inference. I think you're right that it's unlikely to be an issue in practice, given that it's rather unlikely for an existing datatype to even have the correct *kind* to be a `Shape`.